src/harness/reference_models/move_list_calc/move_list.py

```diff
@@ -0,0 +1,594 @@
+        exclusionZone:      a polygon object or a list of locations.
+                            If it is a list of locations, each one being a tuple
+                            with fields 'latitude' and 'longitude'.
+        dpaType:            an enum member of class DpaType
+        hIncAnt:            reference incumbent antenna height (m)
+        numIter:            number of Monte Carlo iterations
+        threshold:          protection threshold (dBm/10 MHz)
+        beamwidth:          protection antenna beamwidth (degree)
+
+    Returns:                the list of indices of grants on the move list for the
+                            given protection constraint
+    """
+
+    # Assign values to the protection constraint
+    constraint = ProtectionConstraint(latitude=protectionPoint.latitude,
+                                      longitude=protectionPoint.longitude,
+                                      lowFrequency=lowFreq,
+                                      highFrequency=highFreq)
+
+    # Identify Nc CBSD grants in the neighborhood of the protection constraint
+    Nc_grants = findGrantsInsideNeighborhood(registrationReq, grantReq, grantInd,
+                                             constraint, exclusionZone, dpaType)
+
+    if len(Nc_grants):  # Found CBSDs in the neighborhood
+
+        # Form the matrix of interference contributions
+        I, sorted_grant_indices, bearings = formInterferenceMatrix(Nc_grants, constraint,
+                                                                hIncAnt, numIter, dpaType)
+
+        # Find the index (nc) of the grant in the ordered list of grants such that
+        # the protection percentile of the interference from the first nc grants is below
+        # the threshold for all azimuths of the receiver antenna.
+        nc = find_nc(I, bearings, threshold, beamwidth)
+
+        # Determine the associated move list (Mc)
+        Mc = sorted_grant_indices[nc:]
+
+        return Mc
+
+    else:
+        
+        return []
+
+
+def findMoveList(protection_specs, protection_points, registration_requests,
+                 grant_requests, num_iter, num_processes, exclusion_zone=None):
+
+    """
+    Main routine to find CBSD indices on the move list.
+
+    Inputs:
+        protection_specs:   a list of protection specifications, a named tuple with
+                            fields of 'lowFreq' (in Hz), 'highFreq' (in Hz),
+                            'antHeight' (in meter), 'beamwidth' (in degree),
+                            'threshold' (in dBm/10MHz)
+        protection_points:  a list of protection points, each one being a tuple with
+                            named fields of 'latitude' and 'longitude'
+        registration_requests: a list of CBSD registration requests, each one being
+                            a dictionary containing CBSD registration information
+        grant_requests:     a list of grant requests, each one being a dictionary
```

Comment:
If a single CBSD has more than one grant, there should be corresponding duplicate entries in the registration list.

Reply:
Hmm, a bit strange (though I see why you did it), but I guess it can be handled in the wrapper without too much difficulty. It might be worth adding a comment here though.

Reply:
This approach was actually based on a suggestion you made to us back in August :).  But I agree with adding a clarifying comment.  We'll do that.

Reply:
I've already forgotten nearly everything that happened in August so I didn't even realize I was being self-contradictory. Sorry about that!